### DISCUSSION TITLE
How to get best image quality when converting .TIF to gray50?

### DISCUSSION THREAD

**Discussion by tcsdmaic** [ImageMagick/ImageMagick#3464] [Help]
We are using ImageMagick via CMD to convert .TIF images to black & white. Meaning we are given a grayscale TIF file with 256 colors, and are converting those to black and white so the output only has 2 colors.

These are the options we're currently using:
pattern:gray50 -compress group4 -density 300

The images come out OK, but sometimes pretty dark. And also text in the file comes out with some blank pixels in it (so black text somehow comes out with some of the blackness removed).

This is an example of how the "before" (grayscale) looks zoomed in on the text and then the "after" (black & white) looks zoomed in on the text.

![grayscale-zoomed in text](https://user-images.githubusercontent.com/17765810/112888036-e02b4a80-90a1-11eb-891c-1c846ac69466.png)
![black white-zoomed in text](https://user-images.githubusercontent.com/17765810/112888039-e02b4a80-90a1-11eb-8eb1-c60522b86c5c.png)

As far as the darkness is concerned, we had some progress using -brightness-contrast to slightly increase the brightness in images, but that didn't help the quality otherwise.

What process / options works best when converting grayscale TIFs to black and white like that?  How can we get the best image quality in the output, so the text isn't all "holey" like that?

**Comment by fmw42** [comment]
What was your exact command line?  What is your exact IM version? Can you provide and example input grayscale TIF that you need converting?

**Reply by tcsdmaic** [reply]
The command line is "magick convert [%inputFile%] -remap pattern:gray50 -compress group4 -density 300 [%outputFile%]", where [%inputFile%] is replaced with the full path of the grayscale TIF and [%outputFile%] is replaced with the full path of the output file.

The Image Magick version I'm using is 7.0.10-35 Q8 x64.

Unfortunately github does not support attaching TIF files.  Suffice it to say the ones we are converting are LZW, grayscale compression and 256 colors.  If you have an image viewer / converter app (we are using [IrfanView](https://www.irfanview.com/)) you can try converting a TIF image to grayscale and use that.

Thank you!

**Comment by fmw42** [comment]
With ImageMagick 7, use magick, not magick convert.  Why are you using Q8? You can zip your tif and upload that. Zip files are allowed.

**Reply by tcsdmaic** [reply]
Why not include "convert"?

Q8 is what I was given.  Is there something bad about it?

OK, I attached a ZIP which contains a grayscale TIF, and then 2 TIFs that were conversions to black & white (with and without "convert").

[ImageMagick-TIFs.zip](https://github.com/ImageMagick/ImageMagick/files/6224243/ImageMagick-TIFs.zip)

**Reply by fmw42** [reply]
Adding convert makes IM use IM 6 syntax rather than IM 7.  Q8 means 8-bit results Q16 would be 16-bit results. So more accurate. But you may not care. The default is Q16. So someone had to make a decision to reduce your Q level.

**Comment by snibgo** [comment]
The OP image has dithering. This means that gray tones are converted to black and white pixels, in the appropriate ratio to simulate the grayness. If you don't want dithering, try "+dither". Or simply threshold at 50%.

The image AllTheThings-grayscale.tif has various shades of gray. Again, you can either dither, or theshold at whatever level you want.

**Reply by tcsdmaic** [reply]
OK...interesting.
Well, I tried my original command line with "+dither" added, but I'm not seeing a difference.

**Reply by fmw42** [reply]
Where did you add +dither? It should come before -remap

**Comment by fmw42** [comment]
Here are a number of different renderings:


```
magick AllTheThings-grayscale.tif[0] -alpha off -remap pattern:gray50 -compress group4 -density 300 AllTheThings-grayscale_fred0.tif


magick AllTheThings-grayscale.tif[0] -alpha off +dither -remap pattern:gray50 -compress group4 -density 300 AllTheThings-grayscale_fred1.tif


magick AllTheThings-grayscale.tif[0] -alpha off -threshold 50% -compress group4 -density 300 AllTheThings-grayscale_fred2.tif


magick AllTheThings-grayscale.tif[0] -alpha off -monochrome -compress group4 -density 300 AllTheThings-grayscale_fred3.tif


magick AllTheThings-grayscale.tif[0] -alpha off -auto-threshold otsu -compress group4 -density 300 AllTheThings-grayscale_fred4.tif

```

[tiff_renderings.zip](https://github.com/ImageMagick/ImageMagick/files/6224702/tiff_renderings.zip)

**Reply by tcsdmaic** [reply]
It is really good to know that where you put an option makes a difference.  I really appreciate all the suggestions...thank you!

I think actually "-brightness-contrast 20x0 -monochrome -compress group4 -density 300" is generally coming out pretty good.
